portal-impl/src/com/liferay/portlet/documentlibrary/util/DLImpl.java

```diff
@@ -873,6 +873,42 @@ public String getUniqueFileName(
 		return getTitleWithExtension(uniqueFileTitle, extension);
 	}
 
+	@Override
+	public String getUniqueFileNameIgnoringTitleExistence(
+		long groupId, long folderId, String fileName) {
+
+		String uniqueFileTitle = FileUtil.stripExtension(fileName);
+
+		String extension = FileUtil.getExtension(fileName);
+
+		int i = 1;
+
+		while (_existsFileEntryByFileName(
+					groupId, extension, folderId, uniqueFileTitle)) {
+
+			uniqueFileTitle = FileUtil.appendParentheticalSuffix(
+				FileUtil.stripExtension(fileName), String.valueOf(i++));
+		}
+
+		return getTitleWithExtension(uniqueFileTitle, extension);
+	}
```

Comment:
Why is the impl. of these methods different than the already existing one? Choose one option (for+break or while loop) and use that in all methods.

Also, the two getUniqueFile* impls are almos identical -- the only difference is that in the original one, an additional condition is checked. Can you merge both implementations? Something like:
```java

	@Override
	public String getUniqueTitle(
		long groupId, long folderId, String title, boolean ignoreDuplicateTitle) {

		String uniqueFileTitle = title;

		int i = 1;

		while ((ignoreDuplicateTitle ||
				!_existsFileEntryByTitle(groupId, folderId, uniqueFileTitle) &&
			  _existsFileEntryByTitle(groupId, folderId, uniqueFileTitle)) {
			
			uniqueFileTitle = FileUtil.appendParentheticalSuffix(
				title, String.valueOf(i));
			i++;
		}

		return uniqueFileTitle;
	}
```
Make the old method use the new, more general, one (I don't like boolean args either, but it is an established pattern).

Also, if you choose to use a while loop, don't use inline post/pre increments. Put them in their own line (as I did in the previous example).

Reply:
thanks!